src/mac/minidump_writer.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{
+    dir_section::{DirSection, DumpBuf},
+    mac::{errors::WriterError, task_dumper::TaskDumper},
+    mem_writer::*,
+    minidump_format::{self, MDMemoryDescriptor, MDRawDirectory, MDRawHeader},
+};
+use std::io::{Seek, Write};
+
+type Result<T> = std::result::Result<T, WriterError>;
+
+pub struct MinidumpWriter {
+    /// The crash context as captured by an exception handler
+    pub(crate) crash_context: crash_context::CrashContext,
+    /// List of raw blocks of memory we've written into the stream. These are
+    /// referenced by other streams (eg thread list)
+    pub(crate) memory_blocks: Vec<MDMemoryDescriptor>,
```

Comment:
This is the only reason that `dump` is `&mut`, but we could do the same thing that the Linux dumper does and just create an internal mut state instead if that signature is problematic.